lib/portus/migrate.rb

```diff
@@ -14,29 +14,37 @@ module Migrate
     #
     # TODO: (mssola) remove in the next version.
     def self.from_humanized_time(duration, default)
-      return duration.send(:minutes) if duration.is_a? Integer
+      return duration.minutes if duration.respond_to? :minutes
 
       # If it's not a String then just return the given default value.
-      return default.send(:minutes) unless duration.is_a? String
+      return default.minutes unless duration.is_a? String
 
-      # If it's a string containing just a number, then convert it and return
-      # it. Otherwise, if it has a bad (old) format, then just return the default.
-      splitted = duration.split(".")
-      if splitted.size == 1
-        val = splitted.first.to_i
-        return val.send(:minutes) unless val == 0
-      end
-      return default.send(:minutes) if splitted.size != 2
+      # If we got an empty String then return the default value.
+      return default.minutes if duration.empty?
 
-      Rails.logger.tagged("deprecated") do
-        Rails.logger.warn "The 'x.minutes' format is deprecated for configuration values such " \
-                          "as `jwt_expiration_time`. From now on these values are expected to " \
-                          "be integers representing minutes."
+      raw_value, method = duration.split('.')
+      value = raw_value.to_i
+
+      # When we found a callable method, notify about deprecation
+      unless method.nil?
+        if value.respond_to? method
+          Rails.logger.tagged("deprecated") do
```

Comment:
Since you are fixing #1470, raise an exception instead of logging it. I'd create a `DeprecationError` exception class, that can be called like this:

```ruby
raise DeprecationError, "Format 'x.minutes' is no longer supported: time values are expected to be given in minutes."
```

This class would add an "[DEPRECATED]" prefix or something like that to the message.

Reply:
While you are at it, do the same for the `self.registry_config` method 😁

Reply:
wanted to do the refactoring first, as it's kinda hard to read right now. Deprecation Exception would be added in another pull request afterwards.

Reply:
btw, is the deprecation only because of the usage of `eval`?